src/FwdState.cc

```diff
@@ -856,7 +856,9 @@ FwdState::connectedToPeer(Security::EncryptorAnswer &answer)
         answer.error.clear(); // preserve error for errorSendComplete()
         if (CachePeer *p = serverConnection()->getPeer())
             peerConnectFailed(p);
-        serverConnection()->close();
+        if (Comm::IsConnOpen(serverConnection()))
```

Comment:
Similar concern here: If PeerConnector was responsible for connecting to the peer, with its own timeouts and closure handlers, why was FwdState keeping serverConnection()? Should not FwdState wait for the EncryptorAnswer to set serverConnection() (and set closeHandler)?

Reply:
The commit b1b730dbb fixes this one too.